src/api/CertificateManagement.ts

```diff
@@ -0,0 +1,184 @@
+                            statusCode: error.response.status,
+                            error: error.response.data as APIError,
+                            returnValue: {} as EnrollmentId,
+                            networkError: false,
+                        })
+                    ) {
+                        return await this.getEnrollmentId(username);
+                    }
+                    return {
+                        returnValue: {} as EnrollmentId,
+                        statusCode: error.response.status,
+                        error: error.response.data as APIError,
+                        networkError: false,
+                    };
+                } else {
+                    return {
+                        returnValue: {} as EnrollmentId,
+                        statusCode: 0,
+                        error: {} as APIError,
+                        networkError: true,
+                    };
+                }
+            });
+    }
+
+    async getEncryptedPrivateKey(username: string): Promise<APIResponse<EncryptedPrivateKey>> {
+        return await this._axios
+            .get(`/certificates/${username}/privateKey`)
+            .then((response: AxiosResponse) => {
+                return {
+                    returnValue: response.data as EncryptedPrivateKey,
+                    statusCode: response.status,
+                    error: {} as APIError,
+                    networkError: false,
+                };
+            })
+            .catch(async (error: AxiosError) => {
+                if (error.response) {
+                    if (
+                        await handleAuthenticationError({
+                            statusCode: error.response.status,
+                            error: error.response.data as APIError,
+                            returnValue: {} as EncryptedPrivateKey,
+                            networkError: false,
+                        })
+                    ) {
+                        return await this.getEncryptedPrivateKey(username);
+                    }
+                    return {
+                        returnValue: {} as EncryptedPrivateKey,
+                        statusCode: error.response.status,
+                        error: error.response.data as APIError,
+                        networkError: false,
+                    };
+                } else {
+                    return {
+                        returnValue: {} as EncryptedPrivateKey,
+                        statusCode: 0,
+                        error: {} as APIError,
+                        networkError: true,
+                    };
+                }
+            });
+    }
+
+    async sendCertificateSigningRequest(
+        username: string,
+        csr: string,
+        encryptedPrivateKey?: EncryptedPrivateKey
+    ): Promise<APIResponse<Certificate>> {
+        let encPrivKey = encryptedPrivateKey != undefined ? encryptedPrivateKey : { iv: "", salt: "", key: "" };
```

Comment:
Is it actually good to default this to an empty key?

Reply:
The API was designed in such a way, that we always require the JSON object to be transmitted. An alternative would be to create an endpoint to register the CSR without an encryptedPrivateKey. 

I agree that the code would be a bit cleaner then. 

@dhelmert @RHelmert we could add an endpoint for this in the next sprint